common/src/test/java/apoc/util/collection/AbstractResourceIterableTest.java

```diff
@@ -0,0 +1,320 @@
+package apoc.util.collection;
+
+import static apoc.util.collection.CollectionTestHelper.asIterator;
+import static apoc.util.collection.CollectionTestHelper.emptyResourceIterator;
+import static apoc.util.collection.CollectionTestHelper.resourceIterator;
+import static apoc.util.collection.ResourceClosingIterator.newResourceIterator;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Stream;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.apache.commons.lang3.mutable.MutableBoolean;
+import org.apache.commons.lang3.mutable.MutableInt;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.neo4j.graphdb.Resource;
+import org.neo4j.graphdb.ResourceIterator;
+
+@RunWith(JUnitParamsRunner.class)
+public class AbstractResourceIterableTest {
+    @Test
+    public void shouldDelegateToUnderlyingIterableForData() {
+        // Given
+        final var iterableClosed = new MutableBoolean(false);
+        final var iteratorClosed = new MutableBoolean(false);
+
+        final var items = Arrays.asList(0, 1, 2);
+
+        // When
+        final var iterable = new AbstractResourceIterable<Integer>() {
+            @Override
+            protected ResourceIterator<Integer> newIterator() {
+                return resourceIterator(items.iterator(), iteratorClosed::setTrue);
+            }
+
+            @Override
+            protected void onClosed() {
+                iterableClosed.setTrue();
+            }
+        };
+        final var iterator = iterable.iterator();
+
+        // Then
+        assertThat( Iterators.asList(iterator)).containsExactlyElementsOf(items);
+        assertThat(iteratorClosed.isTrue()).isTrue();
```

Comment:
This seems a bit repetitive, at least for me. Maybe I would write:
```suggestion
        assertTrue(iteratorClosed.booleanValue());
```
or 
```suggestion
        assertTrue(iteratorClosed.getValue());
```

Same thing for other similar ones (lines 55, 86, 114-117, 167, 194-196, 232-234, 264-266, 295,296, 318).

Reply:
I ended up doing the first suggestion and applied it for the other test files as well